cmd/thv-operator/main.go

```diff
@@ -111,53 +111,88 @@ func main() {
 
 // setupControllersAndWebhooks sets up all controllers and webhooks with the manager
 func setupControllersAndWebhooks(mgr ctrl.Manager) error {
-	// Set up field indexing for MCPServer.Spec.GroupRef
-	if err := mgr.GetFieldIndexer().IndexField(
-		context.Background(),
-		&mcpv1alpha1.MCPServer{},
-		"spec.groupRef",
-		func(obj client.Object) []string {
-			mcpServer := obj.(*mcpv1alpha1.MCPServer)
-			if mcpServer.Spec.GroupRef == "" {
-				return nil
+	// Check feature flags
+	enableServer := isFeatureEnabled("ENABLE_SERVER", true)
+	enableRegistry := isFeatureEnabled("ENABLE_REGISTRY", true)
+	enableAggregation := isFeatureEnabled("ENABLE_AGGREGATION", true)
+
+	// Set up server-related controllers
+	if enableServer {
+		if err := setupServerControllers(mgr, enableRegistry); err != nil {
+			return err
+		}
+	} else {
+		setupLog.Info("ENABLE_SERVER is disabled, skipping server-related controllers")
+	}
+
+	// Set up registry controller
+	if enableRegistry {
+		if err := setupRegistryController(mgr); err != nil {
+			return err
+		}
+	} else {
+		setupLog.Info("ENABLE_REGISTRY is disabled, skipping MCPRegistry controller")
+	}
+
+	// Set up aggregation controllers and webhooks
+	if enableAggregation {
+		if !enableServer {
+			setupLog.Info("ENABLE_AGGREGATION requires ENABLE_SERVER to be enabled, skipping aggregation controllers")
+		} else {
+			if err := setupAggregationControllers(mgr); err != nil {
+				return err
 			}
-			return []string{mcpServer.Spec.GroupRef}
-		},
-	); err != nil {
-		return fmt.Errorf("unable to create field index for spec.groupRef: %w", err)
+		}
+	} else {
+		setupLog.Info("ENABLE_AGGREGATION is disabled, skipping Virtual MCP controllers and webhooks")
 	}
 
+	//+kubebuilder:scaffold:builder
+	return nil
+}
+
+// setupServerControllers sets up server-related controllers (MCPServer, MCPExternalAuthConfig, MCPRemoteProxy, ToolConfig)
+func setupServerControllers(mgr ctrl.Manager, enableRegistry bool) error {
 	// Create a shared platform detector for all controllers
```

Comment:
At least the comment is a bit misleading. I wonder if there is a reason not to pass the detector down from the caller, IIRC it already uses sync.Once?